### PR TITLE
[CARBONDATA-1379] Fixed Date range filter with cast not working

### PR DESCRIPTION
**Pull request by ravipesala**

### CODE REVIEW

**Comment by ravipesala** [comment]
SDV Build Success with Spark 2.1, Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/192/

File: core/src/main/java/org/apache/carbondata/core/keygenerator/directdictionary/timestamp/DateDirectDictionaryGenerator.java [hunk 1]
```diff
@@ -154,14 +147,14 @@ private int generateDirectSurrogateKeyForNonTimestampType(String memberStr) {
...
-    long milli = timeValue + threadLocalLocalTimeZone.get().getOffset(timeValue);
-    return (int) Math.floor((double) milli / MILLIS_PER_DAY) + cutOffDate;
+    return (int) Math.floor((double) timeValue / MILLIS_PER_DAY) + cutOffDate;
   }
 
   public void initialize() {
     if (simpleDateFormatLocal.get() == null) {
       simpleDateFormatLocal.set(new SimpleDateFormat(dateFormat));
       simpleDateFormatLocal.get().setLenient(false);
+      simpleDateFormatLocal.get().setTimeZone(TimeZone.getTimeZone("GMT"));
```
Comment (lionelcao) [inline core/src/main/java/org/apache/carbondata/core/keygenerator/directdictionary/timestamp/DateDirectDictionaryGenerator.java]: Does it mean carbon will process all Date/TimeStamp as GMT?

Comment (CarbonDataQA) [inline core/src/main/java/org/apache/carbondata/core/keygenerator/directdictionary/timestamp/DateDirectDictionaryGenerator.java]: Loading and querying both uses GMT here only for date, we should not be based on machines timezone as it gives issues

**Comment by CarbonDataQA** [comment]
SDV Build Success with Spark 2.1, Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/241/

**Comment by ravipesala** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/647/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**